hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -453,15 +453,15 @@ private void logAuditEvent(boolean succeeded,
 
   private void appendClientPortToCallerContextIfAbsent() {
     final CallerContext ctx = CallerContext.getCurrent();
-    if (isClientPortInfoAbsent(ctx)) {
-      String origContext = ctx == null ? null : ctx.getContext();
-      byte[] origSignature = ctx == null ? null : ctx.getSignature();
-      CallerContext.setCurrent(
-          new CallerContext.Builder(origContext, contextFieldSeparator)
-              .append(CallerContext.CLIENT_PORT_STR, String.valueOf(Server.getRemotePort()))
-              .setSignature(origSignature)
-              .build());
-    }
+    String origContext = ctx == null ? null : ctx.getContext();
+    byte[] origSignature = ctx == null ? null : ctx.getSignature();
+    String clientPort = isClientPortInfoAbsent(ctx) ? CallerContext.CLIENT_PORT_STR :
```

Comment:
It would be nice to have a comment explaining the logic for when this is a Router request, etc.

Reply:
Are we even sure that if it is not a client, it is a router?

Reply:
Yes，it is.  Because when is from dfsrouter ,the callcontext will add clientPort .  The code is in RouterRpcClient.addClientInfoToCallerContext().    In a word， when is comes form dfsrouter , the callcontext has clientPort, or it is not contains clientPort，so we can use isClientPortInfoAbsent()  to distinguish sources

Reply:
> Are we even sure that if it is not a client, it is a router?

This is exactly why I was suggesting to provide better protection for this very condition
https://github.com/apache/hadoop/pull/5552#issuecomment-1507395844

Today, we know that if it is not a client, it is a router. But this is so _fragile_ and can be easily broken if someone other than router also starts using `clientPort` in the context.
The better way is to have a check that only router would pass, something similar to the comment.

Reply:
I add via isdfsrouter now.